controllers/managedocs_controller.go

```diff
@@ -125,6 +129,13 @@ func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			},
 		),
 	)
+	deadMansSnitchSecretPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(meta metav1.Object, _ runtime.Object) bool {
+				return meta.GetName() == r.DeadMansSnitchSecretName
+			},
+		),
+	)
```

Comment:
I believe this is the third predicate that is filtering a secret under a given name, the other 2 are `pagerdutySecretPredicates` and `addonParamsSecretPredicates`. 
Can we merge all of these into a single predicate that will fit a single secret watch? something like `knownSecretsPredicate` which will test for any of the names.

Reply:
Please explain why it is necessary for _this_ specific PR to have to do this refactor.

Reply:
This PR adds another `.watches()` call on the reconciler, for the `Secret` kind, which is a bad practice as it is not really needed. As such, it looks to me like an opportunity to do this very simple refactoring. 

But, if you prefer to push this change, as a separate PR, immediately after this one merges you are welcome to do so. 

The important part is to stop continuously pushing this kind of additions, which can harm the performance of the reconciler, as now we have 3 watches that trigger 3 separate watch pipelines for every secret change in the namespace

Reply:
I would be happy to submit a PR focused on refactoring the code once this one has been merged.

Reply:
Agreed, consider this one resolved